### DISCUSSION TITLE
更新日志 - 22.03.5 - 20230825

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#974] [Announcements]
# 系统
1. 按端口挂载硬盘时防止 USB HUB 连接的设备出现挂载点冲突

# 应用：
* QuickStart：
  1. 修复首页“内网配置”的地址池没立即应用的问题
  3. 修复首页“DNS配置”自定义时可能保存空数据问题，修复某些情况下没有备用DNS输入框的问题 #950

* Jellyfin：
  1. RK系列机型的基础镜像升级到 v10.8.10，在插件页面点击“升级/应用”即可

# 设备
* x86：
  1. 修复上一版本增加的单网口设备自动NAT手动关闭以后又会自动打开的问题 https://github.com/istoreos/istoreos/discussions/947#discussioncomment-6737630
  2. 新增 AX101 WiFi 模块支持 （更正：其实是AX211）
  3. 23.05 固件测试，用于支持N100等12代Intel CPU的硬解。非此固件，需手动升级，详情点击 https://github.com/istoreos/istoreos/issues/965

**Comment by hxpcyj** [comment]
这个版本解决升级后QBittorrent不能启动问题了吗

**Reply by jjm2473** [reply]
重装试试

**Reply by jjm2473** [reply]
如果使用了第三方的软件源，那就不需要反馈了

**Comment by zhenglyu** [comment]
你好 AX200 有希望支持吗？

**Reply by jjm2473** [reply]
intel ax200 应该早就支持了啊，如果你的设备没看到wifi菜单，那么开机以后尽快执行 `dmesg | grep iwlwifi` ，把结果发出来看看

**Reply by colinguohao** [reply]
root@iStoreOS:~# dmesg | grep iwlwifi
[    7.649221] iwlwifi 0000:00:14.3: enabling device (0000 -> 0002)
[    7.649957] iwlwifi: No config found for PCI dev 54f0/0244, rev=0x370, rfid=0x10c000
[    7.653584] iwlwifi: probe of 0000:00:14.3 failed with error -22

**Reply by colinguohao** [reply]
这个是ax101的结果

**Reply by jjm2473** [reply]
我这个也是ax101
<img width="842" alt="image" src="https://github.com/istoreos/istoreos/assets/3896703/595b476e-d4ac-468c-a481-0bec064e852b">

**Reply by jjm2473** [reply]
@colinguohao 你执行下 `lspci -nnk| grep -A3 54f0` 看看

**Reply by colinguohao** [reply]
```
root@iStoreOS:~# lspci -nnk| grep -A3 54f0
lspci: Unable to load libkmod resources: error -12
00:14.3 Network controller [0280]: Intel Corporation Device [8086:54f0]
        DeviceName: Onboard - Ethernet
        Subsystem: Intel Corporation Device [8086:0244]
00:15.0 Serial bus controller [0c80]: Intel Corporation Device [8086:54e8]
```

**Reply by colinguohao** [reply]
![image](https://github.com/istoreos/istoreos/assets/143392300/caaa2ee4-0fc6-473f-9d3a-17defb69a2ce)

**Reply by colinguohao** [reply]
大佬，你这个是ax211吧

**Reply by jjm2473** [reply]
厂商说是ax101啊，而且pcie的设备id也跟你的一样

**Reply by jjm2473** [reply]
不好意思，我拆机以后发现确实是AX211工程样品，厂家说发给我的测试机是集成AX101的，所以我一直以为是AX101。这样看来，目前的iwlwifi还不支持AX101，而是支持AX211。

**Reply by colinguohao** [reply]
你的内核驱动显示出来了，我这个没显示出来........

**Reply by colinguohao** [reply]
ax101需要linux内核更新到6.1.15或以上，然后再打个补丁.......有点折腾
https://tieba.baidu.com/p/8332425004

**Reply by jjm2473** [reply]
因为驱动显示的名称跟实际硬件不符的情况司空见惯了，以为只是识别错误，实际上是厂家给的信息错误。

**Reply by jjm2473** [reply]
那可能要等到23.05了，你可以试试23.05的测试固件，看看内核的日志有没有变化

**Reply by jjm2473** [reply]
23.05使用的wifi子系统就是从6.1.24反向移植的

**Reply by colinguohao** [reply]
![图片](https://github.com/istoreos/istoreos/assets/143392300/155ee75f-0461-4e5b-83bb-88a7b2f87561)

**Reply by colinguohao** [reply]
大佬，23.05刷了，无线部分和前面显示22版本的一样，没加载出来，uname查看这个内核版本也是5.15.125的，这个没错吧？

**Reply by jjm2473** [reply]
没错，你把`dmesg | grep iwlwifi`的结果发一下。

**Reply by colinguohao** [reply]
![图片](https://github.com/istoreos/istoreos/assets/143392300/01141116-210f-451e-b273-00aa324fef25)

**Reply by jjm2473** [reply]
那说明下个版本23.05都还不能支持

**Reply by zhenglyu** [reply]
收到,THKS

**Comment by ssnhd** [comment]
同类插件有的在服务里，有的在 VPN 里，是故意这样设计？

**Reply by jjm2473** [reply]
不是，是插件自己写成这样的，插件想在哪个菜单里面注册都行。
实际上openwrt官方固件根本就没有“VPN”这个菜单，是部分插件自己用了这个菜单。

**Comment by xihesi** [comment]
DHCP/DNS -> 高级设置 ->  关掉 过滤 IPv6 AAAA 记录 功能貌似无效，一样不返回v6地址，望确认

**Reply by jjm2473** [reply]
确认客户端有ipv6地址吗，没有ipv6地址的话，就算返回ipv6地址也不会显示。或者你试试`nslookup taobao.com 114.114.114.114`看看能不能返回ipv6地址，如果可以再尝试`nslookup taobao.com 127.0.0.1`（如果不是在软路由执行，把127.0.0.1改成软路由的ip）

**Reply by xihesi** [reply]
旁路由自身已获得V6地址的，但作为DNS不返回V6，尝试关掉《过滤 IPv6 AAAA 记录》 功能无效, 开启《过滤 IPv4 A 记录》是有用的。
简单来说就是旁路有已获得V6地址，nslookup aliyun.com 255.3.3.3，正常返回v6地址， nslookup aliyun.com 旁路有IP 不返回v6，不使用旁路由IP 作为dns 一切正常，不确定是版本问题还是自身问题，未更新前功能不一样dns V6 正常使用

**Reply by jjm2473** [reply]
那是不是你的上游dns有问题呢？我测试了dns功能没问题
<img width="614" alt="image" src="https://github.com/istoreos/istoreos/assets/3896703/ecb1a383-d3cd-4214-ac7e-f0d3bc802b59">
<img width="409" alt="image" src="https://github.com/istoreos/istoreos/assets/3896703/15c866cd-cab9-4bc7-9d72-a23b46c7c34a">

还有，你是否使用了一些dns过滤工具，例如翻墙插件，去广告插件，有没有试过关掉它们

**Reply by jjm2473** [reply]
还有，你是否自己更新了dnsmasq？

**Reply by xihesi** [reply]
关闭了广告过滤和科学（重置情况下），作为DNS机器来说依旧不返还V6，DNS 设置的上游 无问题，系统是由旧版本更新来的（旧版正常）

上游 dns：192.168.4.111 （自建） 223.5.5.5 （阿里）
旁路由IP：192.168.4.108
上游DNS 可返回v6地址，具体信息如下

```bash
root@iStoreOS:~# cat /tmp/resolv.conf.d/resolv.conf.auto
# Interface lan
nameserver 192.168.4.111
nameserver 223.5.5.5
# Interface lan6
nameserver 192.168.4.111
nameserver 2400:3200::1
root@iStoreOS:~# nslookup taobao.com 192.168.4.108
Server:         192.168.4.108
Address:        192.168.4.108:53

Non-authoritative answer:

Non-authoritative answer:
Name:   taobao.com
Address: 106.11.226.158
Name:   taobao.com
Address: 59.82.122.115

root@iStoreOS:~# nslookup taobao.com
Server:         127.0.0.1
Address:        127.0.0.1:53

Non-authoritative answer:
Name:   taobao.com
Address: 106.11.226.158
Name:   taobao.com
Address: 59.82.122.115

Non-authoritative answer:

root@iStoreOS:~# nslookup taobao.com 192.168.4.111
Server:         192.168.4.111
Address:        192.168.4.111:53

Non-authoritative answer:
Name:   taobao.com
Address: 2408:4002:1f30::a5
Name:   taobao.com
Address: 2408:4001:f00::27c

Non-authoritative answer:
Name:   taobao.com
Address: 106.11.226.158
Name:   taobao.com
Address: 59.82.122.115
```

旁路由 resolv.conf 信息
```bash
root@iStoreOS:~# cat /etc/resolv.conf
search lan
nameserver 127.0.0.1
nameserver ::1
root@iStoreOS:~# nslookup taobao.com
Server:         127.0.0.1
Address:        127.0.0.1:53

Non-authoritative answer:

Non-authoritative answer:
Name:   taobao.com
Address: 106.11.226.158
Name:   taobao.com
Address: 59.82.122.115
```
修改 resolv.conf 后本机可返回v6，但作为DNS 依旧无v6
```bash
root@iStoreOS:~# cat /etc/resolv.conf
search lan
#nameserver 127.0.0.1
nameserver 192.168.4.111
nameserver ::1
root@iStoreOS:~# nslookup taobao.com
Server:         192.168.4.111
Address:        192.168.4.111:53

Non-authoritative answer:
Name:   taobao.com
Address: 2408:4002:1f30::a5
Name:   taobao.com
Address: 2408:4001:f00::27c

Non-authoritative answer:
Name:   taobao.com
Address: 59.82.122.115
Name:   taobao.com
Address: 106.11.226.158

root@iStoreOS:~# nslookup taobao.com localhost
Server:         localhost
Address:        [::1]:53

Non-authoritative answer:
Name:   taobao.com
Address: 106.11.226.158
Name:   taobao.com
Address: 59.82.122.115

Non-authoritative answer:

```
期待查阅指正，谢谢

**Reply by jjm2473** [reply]
我看你是自己配置的一个lan6，而不是使用默认的wan6（注意默认dns是fe80::1，不需要显式配置），既然我测试了dns没问题，那你还是抓包看看吧

**Reply by jjm2473** [reply]
修改/etc/resolv.conf是没用的，只影响软路由自己的dns，而且这个文件会被覆盖

**Reply by jjm2473** [reply]
@xihesi 你恢复出厂设置以后，有关掉“过滤AAAA”的开关吗？

**Comment by wanasaqtemp** [comment]
大佬您好，请问升级之后系统日志里面频繁出现下列信息是什么原因：

Tue Aug 29 19:40:02 2023 daemon.info dnsmasq[1]: exiting on receipt of SIGTERM
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: started, version 2.89 cachesize 1000
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: DNS service limited to local subnets
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: compile time options: IPv6 GNU-getopt no-DBus UBus no-i18n no-IDN DHCP DHCPv6 no-Lua TFTP conntrack ipset no-nftset auth cryptohash DNSSEC no-ID loop-detect inotify dumpfile
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: UBus support enabled: connected to system bus
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq-dhcp[1]: DHCP, IP range 192.168.1.100 -- 192.168.1.249, lease time 12h
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for test
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for onion
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for localhost
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for local
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for invalid
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for bind
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for lan
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: reading /tmp/resolv.conf.d/resolv.conf.auto
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using nameserver 202.96.134.133#53
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using nameserver 202.96.128.166#53
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using nameserver 240e:1f:1::1#53
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for test
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for onion
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for localhost
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for local
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for invalid
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for bind
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: using only locally-known addresses for lan
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: read /etc/hosts - 12 names
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: read /tmp/hosts/dhcp.cfg01411c - 4 names
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq[1]: read /tmp/hosts/odhcpd - 32 names
Tue Aug 29 19:40:04 2023 daemon.info dnsmasq-dhcp[1]: read /etc/ethers - 0 addresses

网络似乎没有中断。

谢谢！

**Reply by jjm2473** [reply]
可能是你的翻墙插件或者去广告插件定时更新规则，更新完后重启了dns服务。具体是什么插件触发的，鬼知道。

**Reply by wanasaqtemp** [reply]
谢谢，好像是SSR+在节点间切换，时间对应上了。

**Comment by ltw546166132** [comment]
最新版本固件iStoreOS 22.03.5 2023082509 有问题啊。内网速度都达不到千兆，我千兆的网口。现在只能700Mbps
![1693415842109](https://github.com/istoreos/istoreos/assets/13818658/53fffdd3-0061-4ae0-8e81-1aadf2af7a56)
![1693415928815](https://github.com/istoreos/istoreos/assets/13818658/430320de-dcd8-444e-b831-8f73bbd641a5)

**Reply by jjm2473** [reply]
你测试的时候内网没有其他大的流量？

**Reply by ltw546166132** [reply]
没有啊，我是刚好昨天重装了下系统，用裸机直接安装的istoreos最新固件。 电脑直接连的软路由lan口测的内网。没什么其他的应用占流量。 我之前用pve虚拟机安装的istoreos 内网测速都能1000m， 感觉是固件问题。 因为上传速度是1000m差不多正常的，下载的速度我测了好多次还是只有700m速度

**Reply by jjm2473** [reply]
你试过其他浏览器吗？

**Reply by ltw546166132** [reply]
试过的，都一样。主要是我重装了最新的固件后才发现的内网变慢。之前测都是1000m，是同一个edge浏览器测的

**Reply by jjm2473** [reply]
那试试前一两个版本的固件，还有执行`lspci -nnk`看看网卡型号

**Reply by ltw546166132** [reply]
你看一下吧  型号是这个
00:1f.4 SMBus [0c05]: Intel Corporation Jasper Lake SMBus [8086:4da3] (rev 01)
        DeviceName: Onboard - Other
        Subsystem: Intel Corporation Device [8086:7270]
00:1f.5 Serial bus controller [0c80]: Intel Corporation Jasper Lake SPI Controller [8086:4da4] (rev 01)
        DeviceName: Onboard - Other
        Subsystem: Intel Corporation Device [8086:7270]
01:00.0 Network controller [0280]: Intel Corporation Wireless 3165 [8086:3165] (rev 81)
        Subsystem: Intel Corporation Dual Band Wireless AC 3165 [8086:4010]
02:00.0 Ethernet controller [0200]: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller [10ec:8168] (rev 15)
        Subsystem: Realtek Semiconductor Co., Ltd. Device [10ec:0123]
        Kernel driver in use: r8169
03:00.0 Ethernet controller [0200]: Realtek Semiconductor Co., Ltd. RTL8111/8168/8411 PCI Express Gigabit Ethernet Controller [10ec:8168] (rev 15)
        Subsystem: Realtek Semiconductor Co., Ltd. Device [10ec:0123]
        Kernel driver in use: r8169

**Comment by vay1314** [comment]
升级完 wan口无法正常拨号，重启后依然如此，回滚之前系统正常。新版的系统日志附上
[log.txt](https://github.com/istoreos/istoreos/files/12495046/log.txt)

**Reply by jjm2473** [reply]
恢复出厂设置试试，你这个根本连网络相关的日志都没有，为什么只有几十秒的日志？还有可能有插件启动卡住了，导致后面的启动脚本运行不了。反正你的系统可以回滚，那不保留配置升级以后测试看看拨号有没有问题。

**Reply by vay1314** [reply]
我这个是升级后重启后的日志，就这么多，没有自动拨号。那只能之后测试下不保留配置升级了。

**Comment by q405812306** [comment]
平台：x86    cpu：n4120
istoreos-21.02.3-2023032001（5.4.166内核，好像是） 升级到 iStoreOS 22.03.5 2023090112  （5.10.176内核） ，docker的容器端口全部对公网开放，而类似GoWebDav这种原来已经开启对公网开放，端口则变成是对公网关闭的。不能保留配置升级，只能重新刷最新固件才行。

之前好像说禁用了docker的防火墙，转交由op防火墙统一管理，这算不算是bug了

**Reply by jjm2473** [reply]
你是不是之前自己升级过dockerd程序？如果没有，固件升级后把`/etc/config/dockerd`，`/etc/config/network`和`/etc/config/firewall`几个文件的内容发出来看看。

**Reply by jjm2473** [reply]
GoWebDav跟docker的防火墙没有任何关系

**Reply by jjm2473** [reply]
我验证了下，确实升级脚本有问题，没把docker的配置清理掉

**Reply by jjm2473** [reply]
哦，是没把docker的升级脚本打包到软件包里

**Reply by jjm2473** [reply]
不过就算升级脚本没运行，那docker的防火墙还是由docker管理啊，原本没开放的端口怎么会开放呢

**Reply by q405812306** [reply]
docker就是用固件自带的，没有升级过。
昨天我直接重新刷掉了，刚刚又验证了一次。这次是端口对公网全部开放，包括容器和GoWebDav的
升级前
![5 4内核](https://github.com/istoreos/istoreos/assets/34687385/5d7cada4-efd6-4895-9953-b7aaa4b741a7)
 升级后
![5 10内核](https://github.com/istoreos/istoreos/assets/34687385/f09bc119-4feb-4e9f-88d0-70c4a4acb3be)


升级前
dockerd配置
```
config globals 'globals'
	option log_level 'warn'
	option iptables '1'
	option data_root '/overlay/upper/opt/docker'

config firewall 'firewall'
	option device 'docker0'
	option extra_iptables_args '--match conntrack ! --ctstate RELATED,ESTABLISHED'
```
network配置
```
config interface 'loopback'
	option device 'lo'
	option proto 'static'
	option ipaddr '127.0.0.1'
	option netmask '255.0.0.0'

config globals 'globals'
	option ula_prefix 'fdc7:97eb:33db::/48'

config interface 'wan'
	option device 'eth0'
	option proto 'pppoe'
	option username 'xxxxxxxxxxx'
	option password 'xxxxxxxxx'
	option ipv6 'auto'

config interface 'wan6'
	option device 'eth0'
	option proto 'dhcpv6'

config device
	option name 'br-lan'
	option type 'bridge'
	list ports 'eth1'
	list ports 'eth2'
	list ports 'eth3'

config interface 'lan'
	option device 'br-lan'
	option proto 'static'
	option netmask '255.255.255.0'
	option ip6assign '60'
	option ipaddr '192.168.2.254'
	option defaultroute '0'

config interface 'docker'
	option device 'docker0'
	option proto 'none'
	option auto '0'
```

升级后
dockerd配置
```
config globals 'globals'
	option log_level 'warn'
	option iptables '1'
	option data_root '/overlay/upper/opt/docker'

config firewall 'firewall'
	option device 'docker0'
	option extra_iptables_args '--match conntrack ! --ctstate RELATED,ESTABLISHED'
```
network配置
```
config interface 'loopback'
	option device 'lo'
	option proto 'static'
	option ipaddr '127.0.0.1'
	option netmask '255.0.0.0'

config globals 'globals'
	option ula_prefix 'fdc7:97eb:33db::/48'

config interface 'wan'
	option device 'eth0'
	option proto 'pppoe'
	option username 'xxxxx'
	option password 'xxxxx'
	option ipv6 'auto'

config interface 'wan6'
	option device 'eth0'
	option proto 'dhcpv6'

config device
	option name 'br-lan'
	option type 'bridge'
	list ports 'eth1'
	list ports 'eth2'
	list ports 'eth3'

config interface 'lan'
	option device 'br-lan'
	option proto 'static'
	option netmask '255.255.255.0'
	option ip6assign '60'
	option ipaddr '192.168.2.254'
	option defaultroute '0'

config interface 'docker'
	option device 'docker0'
	option proto 'none'
	option auto '0'
```

**Reply by jjm2473** [reply]
哦，我想起来了，原版的docker自带防火墙要么全开放，要么全屏蔽，为了能跟op的防火墙配合，在21版的固件里我修改了docker自带防火墙逻辑，22版的时候因为弃用了docker的防火墙，也就没修改docker自带防火墙逻辑了。只是升级脚本没生效，所以老版本固件升级时没把docker自带防火墙给禁用掉。之后的固件会修复升级脚本。

不过gowebdav应该跟这个没任何关系。

**Reply by jjm2473** [reply]
临时方案可以执行这个脚本：

```shell
#!/bin/sh

uci -q set dockerd.globals.iptables=0

uci -q delete firewall.docker.network

DOCKER0_DEVICE=`uci show network | grep -E "^network\\.@device\\[\\d+\\]\\.name='docker0'" | head -n1 | head -c -16`

if [ -n "$DOCKER0_DEVICE" ]; then
    uci delete "$DOCKER0_DEVICE"
fi
uci -q delete network.docker
uci commit

```

**Reply by q405812306** [reply]
好的，谢谢大佬
